cfg/builder/builder_finalize.cc

```diff
@@ -305,106 +320,84 @@ void CFGBuilder::fillInBlockArguments(core::Context ctx, CFG::ReadsAndWrites &Rn
     // This solution is  (|BB| + |symbols-mentioned|) * (|cycles|) + |answer_size| in complexity.
     // making this quadratic in anything will be bad.
 
-    vector<UnorderedSet<core::LocalVariable>> readsByBlock(cfg.maxBasicBlockId);
-    vector<UnorderedSet<core::LocalVariable>> writesByBlock(cfg.maxBasicBlockId);
-    vector<UnorderedSet<core::LocalVariable>> deadByBlock(cfg.maxBasicBlockId);
-
-    for (auto &rds : RnW.reads) {
-        auto &wts = RnW.writes[rds.first];
-        histogramInc("cfgbuilder.readsPerBlock", rds.second.size());
-        if (rds.second.size() == 1 && wts.size() == 1 && *(rds.second.begin()) == *(wts.begin())) {
-            wts.clear();
-            rds.second.clear(); // remove symref that never escapes a block.
-        } else if (wts.empty()) {
-            rds.second.clear();
-        }
-    }
-
-    for (auto &wts : RnW.writes) {
-        histogramInc("cfgbuilder.writesPerBlock", wts.second.size());
-        auto &rds = RnW.reads[wts.first];
-        if (rds.empty()) {
-            wts.second.clear();
-        }
-        for (BasicBlock *bb : rds) {
-            readsByBlock[bb->id].insert(wts.first);
-        }
-        for (BasicBlock *bb : wts.second) {
-            writesByBlock[bb->id].insert(wts.first);
-        }
-    }
-
-    for (auto &dead : RnW.dead) {
-        for (BasicBlock *bb : dead.second) {
-            deadByBlock[bb->id].insert(dead.first);
-        }
-    }
+    const vector<UnorderedSet<core::LocalVariable>> &readsByBlock = RnW.reads;
+    const vector<UnorderedSet<core::LocalVariable>> &writesByBlock = RnW.writes;
+    const vector<UnorderedSet<core::LocalVariable>> &deadByBlock = RnW.dead;
 
     // iterate ver basic blocks in reverse and found upper bounds on what could a block need.
     vector<UnorderedSet<core::LocalVariable>> upperBounds1(cfg.maxBasicBlockId);
     bool changed = true;
-
-    while (changed) {
-        changed = false;
+    {
```

Comment:
Why use scope blocks in this function? Are you trying to trigger destructors, or are they just there for grouping?

I usually prefer dividing sections with comments, but if this is just your personal style that's fine too.